src/odemis/acq/stitching/test/registrar_test.py

```diff
@@ -0,0 +1,295 @@
+            model.MD_DESCRIPTION: u"test",  # tiff doesn't support É (but XML does)
+            model.MD_ACQ_DATE: time.time(),
+            model.MD_BPP: 12,
+            model.MD_BINNING: (1, 2),  # px, px
+            model.MD_PIXEL_SIZE: (1e-6, 2e-5),  # m/px
+            model.MD_POS: (1e-3, -30e-3),  # m
+            model.MD_EXP_TIME: 1.2,  # s
+            model.MD_IN_WL: (500e-9, 520e-9),  # m
+        }
+        intile = model.DataArray(img12, md)
+
+        registrar = IdentityRegistrar()
+        registrar.addTile(intile)
+        
+        
+        self.assertEqual(registrar.getPositions()[0][0], intile.metadata[model.MD_POS])
+
+        # Same thing but with a typical SEM data
+        img8 = np.zeros((256, 356), dtype=np.uint8) + 40
+        md8 = {
+            model.MD_DESCRIPTION: u"test sem",  # tiff doesn't support É (but XML does)
+            model.MD_ACQ_DATE: time.time(),
+            model.MD_PIXEL_SIZE: (1.3e-6, 1.3e-6),  # m/px
+            model.MD_POS: (10e-3, 30e-3),  # m
+            model.MD_DWELL_TIME: 1.2e-6,  # s
+        }
+        intile = model.DataArray(img8, md8)
+        
+        registrar = IdentityRegistrar()
+        registrar.addTile(intile)
+
+        self.assertAlmostEqual(registrar.getPositions()[0][0], intile.metadata[model.MD_POS],5)
+        
+        
+class TestShiftRegistrar(unittest.TestCase):   
+    def test_one_tile(self):
+        """
+        Test that when there is only one tile, it's returned as-is
+        """
+        img12 = np.zeros((2048, 1937), dtype=np.uint16) + 4000
+        md = {
+            model.MD_SW_VERSION: "1.0-test",
+            model.MD_DESCRIPTION: u"test",  # tiff doesn't support É (but XML does)
+            model.MD_ACQ_DATE: time.time(),
+            model.MD_BPP: 12,
+            model.MD_BINNING: (1, 2),  # px, px
+            model.MD_PIXEL_SIZE: (1e-6, 2e-5),  # m/px
+            model.MD_POS: (1e-3, -30e-3),  # m
+            model.MD_EXP_TIME: 1.2,  # s
+            model.MD_IN_WL: (500e-9, 520e-9),  # m
+        }
+        intile = model.DataArray(img12, md)
+
+        registrar = ShiftRegistrar()
+        registrar.addTile(intile)
+        
+        
+        self.assertEqual(registrar.getPositions()[0][0], intile.metadata[model.MD_POS])
+
+        # Same thing but with a typical SEM data
+        img8 = np.zeros((256, 356), dtype=np.uint8) + 40
+        md8 = {
+            model.MD_DESCRIPTION: u"test sem",  # tiff doesn't support É (but XML does)
+            model.MD_ACQ_DATE: time.time(),
+            model.MD_PIXEL_SIZE: (1.3e-6, 1.3e-6),  # m/px
+            model.MD_POS: (10e-3, 30e-3),  # m
+            model.MD_DWELL_TIME: 1.2e-6,  # s
+        }
+        intile = model.DataArray(img8, md8)
+        
+        registrar = ShiftRegistrar()
+        registrar.addTile(intile)
+
+        self.assertAlmostEqual(registrar.getPositions()[0][0], intile.metadata[model.MD_POS],5)
+        
+        
+    def test1(self):
```

Comment:
Give better names to the test functions. Ex: test_shift_real(). Also, some of the test cases are so similar, that you probably could just put them together in a loop. For instance, vary numTiles, overlap, and method, and check that the detected position is correct.

BTW, on the one hand, numTiles = 2 currently means that there is just one tile with a little bit of jitter. It's very little "test". On the other hand, the expectation that the registrar detects the original position within 1 pixel is too strong. IMHO, it'd be fine if the detected position is within 10% of the overlap. That might even be still a very strong expectation on large tilings, as the position errors can get accumulated as you go away from the first tile.